compiler/ast/zed/type.go

```diff
@@ -0,0 +1,66 @@
+package zed
```

Comment:
I think `zed` is a confusing name for this given that that's also the name of the overall system and the repository (and thus an ancestor package, too).

How about moving the types in `compiler/ast` down into `compiler/ast/api` and then moving the `compiler/ast/zed` contents up a level to `compiler/ast`?

Reply:
Zed is overloaded but it means the data model too.  I thought the meaning was clear on context.  I'm open to changing it but calling as AST an "api" doesn't make sense to me.  Moreover, they will no longer be used in the service API.

Reply:
When I think of a package name, I always think about the name as a prefix of the entities inside.  So , zed.Type makes perfect sense.  That's what it is.  It's a common type across ZSON, ZNG, and ZST conforming to the Zed data model.  My concern was that we might create another package called zed that is an API to the Zed world.  This could live in the root of the repo, and if that were the case, the type and value definitions in compiler/ast/zed could live there instead.

Reply:
> Zed is overloaded but it means the data model too. I thought the meaning was clear on context.

It's clear to me, but I imagine someone new to the code will struggle to remember that `zed.Type` and `zed.Value` are AST objects while `zng.Type` and `zng.Value` are runtime objects, and I think better package names would help. But this is something we can address later, so let's merge what's here.

>  but calling as AST an "api" doesn't make sense to me. Moreover, they will no longer be used in the service API.

Yeah, that was a poor proposal!